milli/src/update/settings.rs

```diff
@@ -283,6 +298,55 @@ impl<'a, 't, 'u, 'i> Settings<'a, 't, 'u, 'i> {
         }
     }
 
+    fn update_synonyms(&mut self) -> anyhow::Result<bool> {
+        match self.synonyms {
+            Setting::Set(ref synonyms) => {
+                let old_synonyms = self.index.synonyms(self.wtxn)?.unwrap_or_default();
+
+                let mut config = AnalyzerConfig::default();
+
+                let stop_words = self.index.stop_words(self.wtxn)?;
+                if let Some(stop_words) = &stop_words {
+                    config.stop_words(stop_words);
+                }
+
+                let analyzer = Analyzer::new(config);
+
+                let normalize = |text: &String| {
+                    analyzer
+                        .analyze(text)
+                        .tokens()
+                        .filter_map(|token|
+                            if token.is_word() { Some(token.text().to_string()) } else { None }
+                        )
+                        .collect::<Vec<_>>()
+                };
+
+                let new_synonyms = synonyms
+                    .iter()
+                    .map(|(word, synonyms)| {
+                        let normalized_word = normalize(word);
+                        let normalized_synonyms = synonyms.iter()
+                            .map(normalize)
+                            .unique()
```

Comment:
I am not sure that using unique here is a good thing, we want to merge the synonyms instead of discarding the duplicates.

Reply:
what do you mean by merging? I think when user inserts synonyms `["New York", "New york"]` we want to store just `["new york"]` (roughly speaking) which is in fact a duplicates discarding.

Reply:
Hum, yeah, sorry I was thinking about `{ "a": ["x"], "A": ["Y"] }` we want to get `{ "a": ["x", "y"] }`.

Reply:
oh yeah got it. That's need to be fixed too, thx.